src/app/design/frontend/base/default/template/payment/form/omisecc.phtml

```diff
@@ -1,77 +1,119 @@
-
-                            jQuery("#payment-buttons-container").find("button").prop("disabled", false);
-                        } else if (typeof response.card != 'undefined' && !response.card.security_code_check) {
-                            // Display an error message - "Omise Response: Card authorization failure.".
-                            formValidation.css({'opacity': 1, 'display': 'block'});
-                            formValidationMsg.html("Omise Response: Card authorization failure.");
-
-                            jQuery("#payment-buttons-container").find("button").prop("disabled", false);
-                        } else {
-                            form.find("[data-omise=holder_name]").prop("disabled", true);
-                            form.find("[data-omise=number]").prop("disabled", true);
-                            form.find("[data-omise=expiration_month]").prop("disabled", true);
-                            form.find("[data-omise=expiration_year]").prop("disabled", true);
-                            form.find("[data-omise=security_code]").prop("disabled", true);
-
-                            form.find("[data-omise=omise_token]").val(response.id);
-
-                            return superMethod();
-                        };
-                    });
+    if ((typeof jQuery === 'undefined') && ! window.jQuery) {
+
+        if (typeof $ == 'function') {
+            var dlSignIsUsing = true;
+        } else {
+            var dlSignIsUsing = false;
+        }
+
+        function getScript(url, success) {
+            var script     = document.createElement('script');
+                script.src = url;
+
+            var head = document.getElementsByTagName('head')[0],
+                done = false;
+
+            // Attach handlers for all browsers
+            script.onload = script.onreadystatechange = function() {
+                if (! done && (!this.readyState || this.readyState == 'loaded' || this.readyState == 'complete')) {
+                    done = true;
 
-                });
+                    // callback function provided as param
+                    success();
+
+                    script.onload = script.onreadystatechange = null;
+                    head.removeChild(script);
+                };
+            };
+
+            head.appendChild(script);
+        };
+
+        getScript('https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js', function() {
+            $nj = jQuery.noConflict();
+            hookPaymentSave($nj);
+        });
+    } else {
+        $nj = jQuery.noConflict();
+        hookPaymentSave($nj);
+    }
+
+    function hookPaymentSave(jQuery) {
+        Payment.prototype.save = Payment.prototype.save.wrap(function(superMethod) {
+            var paymentMethod = jQuery("#co-payment-form").find("input[name='payment[method]']:checked").val();
+
+            if (typeof paymentMethod == "undefined") {
```

Comment:
@nimid good advice, I also like the way to move `else` condition to the top of the code block.
```js
if (typeof paymentMethod == "undefined") {
    alert('Please select one method');
    return false;
}

if (paymentMethod != "omise_gateway") {
    return superMethod();
}

var omiseValidation = new Validation('co-payment-form');

// If validate pass
if (omiseValidation.validate()) {
  // Take care only success case.
}
```

👍 

will fix it